valohai_yaml/pipelines/conversion.py

```diff
@@ -55,22 +55,24 @@ def convert_deployment_node(self, node: DeploymentNode) -> ConvertedObject:
     def convert_executionlike_node(self, node: Union[ExecutionNode, TaskNode]) -> ConvertedObject:
         node_data = node.serialize()
         step_name = node_data.pop("step")
-        override = node_data.pop("override", {})
+
         step = self.config.get_step_by(name=step_name)
         if not step:  # pragma: no cover
             raise ValueError(f"Step {step_name} not found in {self.config}")
         step_data = step.serialize()
-        step_data.update(override)
+        override = node_data.pop("override", None)
 
-        parameters_from_node = node.get_parameter_defaults()
-        parameters_from_step = step.get_parameter_defaults(include_flags=True)
-        step_data["parameters"] = parameters_from_step
-        step_data["parameters"].update(parameters_from_node)
+        all_inputs = override.pop("inputs", []) + step_data.pop("inputs", [])
+        merged_inputs = [next(key for key in all_inputs if key["name"] == name) for name in step.inputs]
+        step_data['inputs'] = {input['name']: listify(input.get('default', None)) for input in merged_inputs}
```

Comment:
I'm not following what's happening in this code at all 😅

Reply:
Code takes a list of possible names of inputs/parameters from step, then simply takes first item of all_(array) with the same name. I should probably rename vars for the good